core/src/test/java/com/linecorp/armeria/internal/ContextFutureCallbackArgumentsProvider.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+
+import com.google.common.util.concurrent.MoreExecutors;
+
+public class ContextFutureCallbackArgumentsProvider implements ArgumentsProvider {
+
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+        final Arguments thenApply = Arguments.of(
+                (BiConsumer<CompletableFuture<?>, AtomicBoolean>) (future, called) -> {
+                    future.thenApply(res -> {
+                        called.set(true);
+                        return null;
+                    }).exceptionally(cause -> {
+                        called.set(true);
```

Comment:
This is not called because in `exceptionally`, pushing with a wrong context happens as well.
I'm wondering we should mitigate this by not pushing the context in `exceptionally`. 🤔

Reply:
That's a good question. :thinking: 

It is also possible that a user tries to handle an exception using `handle` or `whenComplete`, so I think it's not a good idea to handle `exceptionally` exceptionally. (pun intended :laughing:) i.e. I can't think of a good solution other than just logging.

Reply:
😆